src/OpenTelemetry.Extensions/TracerProviderBuilderExtensions.cs

```diff
@@ -53,14 +54,42 @@ public static class TracerProviderBuilderExtensions
     /// Adds the <see cref="BaggageActivityProcessor"/> to the <see cref="TracerProviderBuilder"/>.
     /// </summary>
     /// <param name="builder"><see cref="TracerProviderBuilder"/> to add the <see cref="BaggageActivityProcessor"/> to.</param>
+    /// <param name="baggageKeyPredicate">Predicate to determine which baggage keys should be added to the activity.</param>
     /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
     public static TracerProviderBuilder AddBaggageActivityProcessor(
-        this TracerProviderBuilder builder)
+        this TracerProviderBuilder builder,
+        Predicate<string> baggageKeyPredicate)
     {
         Guard.ThrowIfNull(builder);
+        Guard.ThrowIfNull(baggageKeyPredicate);
+
+#pragma warning disable CA2000 // Dispose objects before losing scope
+        return builder.AddProcessor(new BaggageActivityProcessor(baggageKeyPredicate));
+#pragma warning restore CA2000 // Dispose objects before losing scope
+    }
+
+    /// <summary>
+    /// Adds the <see cref="BaggageActivityProcessor"/> to the <see cref="TracerProviderBuilder"/>.
+    /// </summary>
+    /// <param name="builder"><see cref="TracerProviderBuilder"/> to add the <see cref="BaggageActivityProcessor"/> to.</param>
+    /// <param name="expr">Regular expression to determine which baggage keys should be added to the activity.</param>
+    /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+    public static TracerProviderBuilder AddBaggageActivityProcessor(
+        this TracerProviderBuilder builder, string expr)
+    {
+        Guard.ThrowIfNull(builder);
+        Guard.ThrowIfNullOrEmpty(expr);
+
+        if (expr == "*")
```

Comment:
Is `*` a valid regex? This API seems a little awkward. Instead of forcing users to pass "*" as regex when they want everything why not just give them a default overload. Something like this...

```csharp
    // Captures all baggage
    public static TracerProviderBuilder AddBaggageActivityProcessor(this TracerProviderBuilder builder) {}

    // Captures baggage based on key predicate
    public static TracerProviderBuilder AddBaggageActivityProcessor(this TracerProviderBuilder builder, Predicate<string> baggageKeyPredicate) {}

    // Captures baggage based on key regex
    public static TracerProviderBuilder AddBaggageActivityProcessor(this TracerProviderBuilder builder, string baggageKeyRegex) {}
```

Personally, I wouldn't offer the regex overload. We have the `Predicate<string>` one users may implement that using a regex if they want, right?